saleor/graphql/product/bulk_mutations/products.py

```diff
@@ -706,23 +711,40 @@ def perform_mutation(cls, root, info, **data):
             warehouses = cls.get_nodes_or_error(
                 warehouse_ids, "warehouse", only_type=Warehouse
             )
-            cls.update_or_create_variant_stocks(variant, stocks, warehouses)
+
+            plugins_manager = info.context.plugins
+            cls.update_or_create_variant_stocks(
+                variant, stocks, warehouses, plugins_manager
+            )
 
         variant = ChannelContext(node=variant, channel_slug=None)
         return cls(product_variant=variant)
 
     @classmethod
     @traced_atomic_transaction()
-    def update_or_create_variant_stocks(cls, variant, stocks_data, warehouses):
+    def update_or_create_variant_stocks(
+        cls, variant, stocks_data, warehouses, plugins_manager
+    ):
+        variant_for_webhook = get_product_variant_data_from_warehouses(
+            variant, warehouses
+        )
+
         stocks = []
         for stock_data, warehouse in zip(stocks_data, warehouses):
-            stock, _ = warehouse_models.Stock.objects.get_or_create(
+            stock, is_created = warehouse_models.Stock.objects.get_or_create(
                 product_variant=variant, warehouse=warehouse
             )
             stock.quantity = stock_data["quantity"]
             stocks.append(stock)
         warehouse_models.Stock.objects.bulk_update(stocks, ["quantity"])
 
+        trigger_product_variant_back_in_stock_webhook(
+            variant_for_webhook, plugins_manager
+        )
+        trigger_product_variant_out_of_stock_webhook(
+            variant_for_webhook, plugins_manager
+        )
```

Comment:
We are triggering both webhooks in that method because we can create a new stock with a value bigger than 0 (case for back_in_stock) or update the existing stock object to 0 (case for out_of_stock).
But each trigger method is not equivalent to the manager method - under these functions we have some checks if webhook can be triggered or not.
I will change the names of these functions because they might missleading.